Panic on collector start when using otlp exporter with `tls: insecure: true` and `headers_setter` extension

### Component(s)

extension/headerssetter

### What happened?

## Description
When using the headers_setter extension with an otlp exporter, configured with `tls: insecure: true`, the contrib collector panics on start. Removing the `tls: insecure: true` or the `headers_setter` extension from the exporter config, allows start up to succeed. 

## Steps to Reproduce
Using the attached config.yaml, run:
```
docker run -v $(pwd)/config.yaml:/etc/otelcol-contrib/config.yaml otel/opentelemetry-collector-contrib:0.66.0
```

## Expected Result
The collector starts and uses context to attach a header to all traces exported

## Actual Result
The collector crashes and does not start

### Collector version

v0.66.0

### Environment information

## Environment
OS: OSX 12.6.1
Docker Desktop 4.13.1

### OpenTelemetry Collector configuration

```yaml
extensions:
  headers_setter:
    headers:
      - key: X-Scope-OrgID
        from_context: tenant_id
      - key: User-ID
        value: user_id

receivers:
  otlp:
    protocols:
      http:
        include_metadata: true

exporters:
  otlp:
    endpoint: otelcol:4317
    tls:
      insecure: true
    auth:
      authenticator: headers_setter

service:
  extensions: [ headers_setter ]
  pipelines:
    traces:
      receivers: [ otlp ]
      exporters: [ otlp ]
```


### Log output

```shell
» docker run -v $(pwd)/config.yaml:/etc/otelcol-contrib/config.yaml otel/opentelemetry-collector-contrib:0.66.0
2022/11/28 23:06:23 proto: duplicate proto type registered: jaeger.api_v2.PostSpansRequest
2022/11/28 23:06:23 proto: duplicate proto type registered: jaeger.api_v2.PostSpansResponse
2022-11-28T23:06:23.263Z        info    service/telemetry.go:111        Setting up own telemetry...
2022-11-28T23:06:23.263Z        info    service/telemetry.go:141        Serving Prometheus metrics      {"address": ":8888", "level": "Basic"}
2022-11-28T23:06:23.264Z        info    service/service.go:89   Starting otelcol-contrib...     {"Version": "0.66.0", "NumCPU": 6}
2022-11-28T23:06:23.264Z        info    extensions/extensions.go:41     Starting extensions...
2022-11-28T23:06:23.264Z        info    extensions/extensions.go:44     Extension is starting...        {"kind": "extension", "name": "headers_setter"}
2022-11-28T23:06:23.264Z        info    extensions/extensions.go:48     Extension started.      {"kind": "extension", "name": "headers_setter"}
2022-11-28T23:06:23.264Z        info    pipelines/pipelines.go:74       Starting exporters...
2022-11-28T23:06:23.264Z        info    pipelines/pipelines.go:78       Exporter is starting... {"kind": "exporter", "data_type": "traces", "name": "otlp"}
2022-11-28T23:06:23.264Z        info    service/service.go:115  Starting shutdown...
2022-11-28T23:06:23.264Z        info    pipelines/pipelines.go:118      Stopping receivers...
2022-11-28T23:06:23.264Z        info    pipelines/pipelines.go:125      Stopping processors...
2022-11-28T23:06:23.264Z        info    pipelines/pipelines.go:132      Stopping exporters...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x550fa0]

goroutine 1 [running]:
google.golang.org/grpc.(*ClientConn).Close(0x0)
        google.golang.org/grpc@v1.51.0/clientconn.go:1016 +0x30
go.opentelemetry.io/collector/exporter/otlpexporter.(*exporter).shutdown(0xffffffffffffffff?, {0x5f09247?, 0x4?})
        go.opentelemetry.io/collector/exporter/otlpexporter@v0.66.0/otlp.go:93 +0x20
go.opentelemetry.io/collector/component.ShutdownFunc.Shutdown(...)
        go.opentelemetry.io/collector/component@v0.66.0/component.go:95
go.opentelemetry.io/collector/exporter/exporterhelper.newBaseExporter.func2({0x6bd7500, 0x40000b6018})
        go.opentelemetry.io/collector@v0.66.0/exporter/exporterhelper/common.go:182 +0x60
go.opentelemetry.io/collector/component.ShutdownFunc.Shutdown(...)
        go.opentelemetry.io/collector/component@v0.66.0/component.go:95
go.opentelemetry.io/collector/service/internal/pipelines.(*Pipelines).ShutdownAll(0x4000b92be0, {0x6bd7500, 0x40000b6018})
        go.opentelemetry.io/collector@v0.66.0/service/internal/pipelines/pipelines.go:135 +0x294
go.opentelemetry.io/collector/service.(*service).Shutdown(0x4000b7d080, {0x6bd7500, 0x40000b6018})
        go.opentelemetry.io/collector@v0.66.0/service/service.go:121 +0xd0
go.opentelemetry.io/collector/service.(*Collector).shutdownServiceAndTelemetry(0x40010eba70, {0x6bd7500?, 0x40000b6018?})
        go.opentelemetry.io/collector@v0.66.0/service/collector.go:264 +0x30
go.opentelemetry.io/collector/service.(*Collector).setupConfigurationComponents(0x40010eba70, {0x6bd7500, 0x40000b6018})
        go.opentelemetry.io/collector@v0.66.0/service/collector.go:166 +0x1fc
go.opentelemetry.io/collector/service.(*Collector).Run(0x40010eba70, {0x6bd7500, 0x40000b6018})
        go.opentelemetry.io/collector@v0.66.0/service/collector.go:190 +0x30
go.opentelemetry.io/collector/service.NewCommand.func1(0x400056f200, {0x5f1280e?, 0x2?, 0x2?})
        go.opentelemetry.io/collector@v0.66.0/service/command.go:53 +0x3b8
github.com/spf13/cobra.(*Command).execute(0x400056f200, {0x40000b0220, 0x2, 0x2})
        github.com/spf13/cobra@v1.6.1/command.go:916 +0x5e0
github.com/spf13/cobra.(*Command).ExecuteC(0x400056f200)
        github.com/spf13/cobra@v1.6.1/command.go:1044 +0x368
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.6.1/command.go:968
main.runInteractive({{0x40009fc360, 0x40009fd530, 0x40009fc780, 0x40009fc000}, {{0x5f3b360, 0xf}, {0x5fb82c8, 0x1f}, {0x5f0c283, 0x6}}, ...})
        github.com/open-telemetry/opentelemetry-collector-releases/contrib/main.go:32 +0x40
main.run(...)
        github.com/open-telemetry/opentelemetry-collector-releases/contrib/main_others.go:11
main.main()
        github.com/open-telemetry/opentelemetry-collector-releases/contrib/main.go:25 +0x124
```


### Additional context

Context is that I am attempting to use the `headers_setter` extension to add a multi-tenancy header to the otlp grpc exporter pointing at a tempo backend, as mentioned in https://grafana.com/docs/tempo/latest/configuration/multitenancy/

--- a/.chloggen/header-setter-transport-security.yaml
+++ b/.chloggen/header-setter-transport-security.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: headerssetter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Do not require the secure transport for the headers setter extension.
+
+# One or more tracking issues related to the change
+issues: [16508]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/extension/headerssetterextension/extension.go
+++ b/extension/headerssetterextension/extension.go
@@ -89,9 +89,11 @@ func (h *headersPerRPC) GetRequestMetadata(
 	return metadata, nil
 }
 
-// RequireTransportSecurity always returns true for this implementation.
+// RequireTransportSecurity always returns false for this implementation.
+// The header setter is not sending auth data, so it should not require
+// a transport security.
 func (h *headersPerRPC) RequireTransportSecurity() bool {
-	return true
+	return false
 }
 
 // headersRoundTripper intercepts downstream requests and sets headers with

